### ISSUE TITLE
Misterious $folded: appears

### ISSUE THREAD

**Issue by crystalfp** [swagger-api/swagger-editor#742]
Just started using Swagger editor: fantastic! I'm working on windows using Firefox.
I use the definition section to factorize the same response from two methods this way:

```
definitions:
  Annotation:
    type: object
    properties:
      id:
        type: string
      title:
        type: string
      text:
        type: string
        description: Content of the annotation
```

And in both methods I refer to it using:

```
      responses:
        '200':
          description: Successful response
          schema:
            $ref: '#/definitions/Annotation'
```

Now only for the first method in the preview panel a mysterious field called $folded: appears after 'test'.
Also if I push the two arrows button to switch to json, the field becomes: `$folded: false`
What is it? Seems it appears after I collapsed and expanded my methods in the preview panel.
To make it disappears I have to load one of the examples and then reload my file.
Thanks!
mario

**Comment by steven10172** [comment]
I had the same issue. After cutting and pasting the responses section `$folded` went away.

**Comment by pratheekhegde** [comment]
it appears suddenlty, i sometimes refresh my swagger-editor. or cut and paste as told above.

**Comment by rmalayter** [comment]
Seeing the same as of this date at the live demo editor http://editor.swagger.io/#/

**Comment by jasontrublu** [comment]
same issue, does not disappear when cutting&pasting

**Comment by saharj** [comment]
I'm not able to reproduce this issue, I need precise steps.

**Comment by rmalayter** [comment]
@saharj It appears non-deterministic. I just re-opened the same file that had the issue before and it is no longer there. Where does $folded come from in the editor code, and what does it mean? That might help us generate reproduction steps.

**Comment by saharj** [comment]
I'm going to close this issue for now. If you see the bug again, please let us know and we'll reopen the issue. 
It would be helpful to have precise steps to reproduce the issue.

**Comment by SwiTool** [comment]
Confirmed:
![](http://puu.sh/nBqkn/eee0029f11.png "$folded bug")

For me, it happened when I first wrote responses schemas, and then the definitions.

**Comment by takahashiyuya** [comment]
I have the same issue also.

**Comment by saharj** [comment]
@takahashiyuya Can you please tell us how you get the issue? I am unable to reproduce this

**Comment by takahashiyuya** [comment]
@saharj I'm sorry I don't know correct procedure. However the issue occurred when I repeat the copy-and-paste.

**Comment by rmalayter** [comment]
@saharj I think it is really hard to get a repo case because it appears "random" to the user. Which could means it is based on timing of callbacks firing or completing. I couldn't find the string `$folded` anywhere in the code (might not have been looking in right place/ranch), so it could be generating that `$folded` string itself somehow. I saw it randomly in both Chrome/Win10 and Firefox/Linux.

**Comment by SMissot** [comment]
I have the same issue with the Swagger Editor.
Running latest version of chrome on W10

After refreshing, the $folded disappears

**Comment by tommyjcarpenter** [comment]
This is happening to me as well using the Swagger editor.

**Comment by victorbadilaventivo** [comment]
I have the same issue

**Comment by IS-smcleod** [comment]
Looks like bleed over from the GUI's folding.  I imagine which levels are folded is stored in the parsed object.  Most likely to allow the two side to synchronize what is folded.  Then the code that displays the examples is picking it up and displaying it.

You should be able to reproduce this with any file that has multiple examples/definitions by collapsing and expanding a parent level, like the operation.  I tried to reproduce it with the new file template but it didn't happen until there was more than one example.

TL;DR

Load the attached file, collapse and expand the get /persons operation, $folded should be at the end of the first response example.

[swagger.txt](https://github.com/swagger-api/swagger-editor/files/351081/swagger.txt)

**Comment by rmalayter** [comment]
@IS-smcleod great work finding a repro case, which I couldn't do. However, I can't can't reproduce using http://editor.swagger.io/#/ with the latest Chrome and your test file.  I wonder if it was fixed by someone already?

**Comment by IS-smcleod** [comment]
@rmalayter I was able to reproduce the problem with http://editor.swagger.io/#/.  I thought it might just be a firefox issue because I was testing with Firefox 47.0.1 but I also got the bug with Chrome 51.0.2704.103 and 51.0.2704.106 (since it updated when I checked the version).

![screenshot](https://cloud.githubusercontent.com/assets/8934164/16785449/fe01999e-4842-11e6-94c1-e85932551c35.png)

**Comment by miqui** [comment]
hi, was able to reproduce (i.e. "see") the $folded issue. A refresh/re-paste "made it go away" - my env is win10 x64 with Chrome 52.0.2743.116

**Comment by NeoElit** [comment]
Same issue on firefox 50.1.0 on ubuntu 16.04LTS

**Comment by zurfluh** [comment]
I have the `$folded` properties in Chrome 57.0.2987.133 / Windows 10 (64-bit)

after I collapsed and expanded one _path_ that has one _get_ and one _put_ action.

![image](https://cloud.githubusercontent.com/assets/2371518/25124827/2f793470-242d-11e7-8acb-306fb5d21f04.png)

**Comment by BhavikSheth** [comment]
I just got this error now while working. Swagger giving "Object
type: "string" $folded: false" error. 

My console is giving "No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:57555' is therefore not allowed access. The response had HTTP status code 502. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled." error.

I am building a web application using ReactJS and both errors started at the same time.

**Comment by webron** [comment]
I'm afraid we don't have the capacity to solve issues in older versions of the editor. Try the latest version, and if you have other issues, please open new tickets.

**Comment by luhaosjs** [comment]
if no errors happened in your editor, just refresh
